raiden/utils/nursery.py

```diff
@@ -130,14 +139,20 @@ def subprocess_stopped(result: AsyncResult) -> None:
             def spawn_under_watch(function: Callable, *args: Any, **kwargs: Any) -> Greenlet:
                 greenlet = gevent.spawn(function, *args, **kwargs)
 
+                def kill_grenlet(_result: AsyncResult) -> None:
+                    # gevent.GreenletExit must **not** be used here, since that
+                    # exception is considered a successful run and it is not
+                    # re-raised on calls to `get`
+                    exception = RuntimeError("Janitor is stopping")
+                    greenlet.throw(exception)
+
                 # The Event.rawlink is executed inside the Hub thread, which
                 # does validation and *raises on blocking calls*, to go around
                 # this a new greenlet has to be spawned, that in turn will
                 # raise the exception.
-                def spawn_to_kill() -> None:
-                    gevent.spawn(greenlet.throw, gevent.GreenletExit())
+                callback = SpawnedLink(kill_grenlet)
 
-                janitor._stop.rawlink(lambda _stop: spawn_to_kill())
+                janitor._stop.rawlink(callback)
```

Comment:
Why not use regular `.link` instead of `SpawnedLink` and `.rawlink`?

Reply:
just followed the style used in a few lines aboev

Reply:
just to elaborate a bit, not all objects have `link`, I believe that `AsyncResult`s only have `rawlink`, therefore in some places this is required. I Believe this was the reason for the `rawlink` above, here `_stop` is an `Event` which may have `link` (I'm not sure, would have to check), anyways, `rawlink` and `link` both works if appropriate care is taken.